pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,304 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package logverbosity
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	LOG_VERBOSITY = "log-verbosity"
+)
+
+type Command struct {
+	clientConfig clientcmd.ClientConfig
+	command      string
+}
+
+var (
+	virtAPIFlag        bool
+	virtControllerFlag bool
+	virtHanderFlag     bool
+	virtLauncherFlag   bool
+	virtOperatorFlag   bool
+	allFlag            bool
+	resetFlag          bool
+)
+
+func NewCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "log-verbosity",
+		Short:   "Set/Reset log verbosity level",
+		Long:    "Set log verbosity level to one or more compoents (virt-api/virt-controller/virt-handler/virt-llauncher/virt-operator). Reset the log verbosity level to all components.",
+		Example: usage(),
+		Args:    cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := Command{command: LOG_VERBOSITY, clientConfig: clientConfig}
+			return c.RunE(args)
+		},
+	}
+
+	cmd.Flags().BoolVar(&virtAPIFlag, "virt-api", virtAPIFlag, "Specify virt-api log verbosity level.")
+	cmd.Flags().BoolVar(&virtControllerFlag, "virt-controller", virtControllerFlag, "Specify virt-controller log verbosity level.")
+	cmd.Flags().BoolVar(&virtHanderFlag, "virt-handler", virtHanderFlag, "Specify virt-handler log verbosity level.")
+	cmd.Flags().BoolVar(&virtLauncherFlag, "virt-launcher", virtLauncherFlag, "Specify virt-launcher log verbosity level.")
+	cmd.Flags().BoolVar(&virtOperatorFlag, "virt-operator", virtOperatorFlag, "Specify virt-operator log verbosity level.")
+	cmd.Flags().BoolVar(&allFlag, "all", allFlag, "Set all component log verbosity levels.")
+	cmd.Flags().BoolVar(&resetFlag, "reset", resetFlag, "Reset log verbosity level.")
+
+	// no required flag
+
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func usage() string {
+	usage := "  # Set a log verbosity level to a component:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --virt-api"
+	usage += "  # Set a log verbosity level to multiple components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --virt-api  --virt-handler"
+	usage += "  # Set a log verbosity level to all components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity 3 --all"
+	usage += "  # Reset the log verbosity level to all components:\n"
+	usage += "  {{ProgramName}} adm logVerbosity --reset"
+
+	return usage
+}
+
+func (c *Command) RunE(args []string) error {
+	/*
+	 * flag strength: reset > all > component
+	 */
+
+	var logLevel uint
+
+	// parse argument
+	if !resetFlag {
```

Comment:
this is a complex series of is statement, I think it can be replaced with a switch case.

Reply:
I'll look into your suggestion.